src/Widgets/WallpaperContainer.vala

```diff
@@ -120,16 +124,38 @@ public class WallpaperContainer : Gtk.FlowBoxChild {
         overlay.add (card_box);
         overlay.add_overlay (check_revealer);
 
+        var event_box = new Gtk.EventBox ();
+        event_box.add (overlay);
+
         halign = Gtk.Align.CENTER;
         valign = Gtk.Align.CENTER;
-
         margin = 6;
-        add (overlay);
+        add (event_box);
+
+        if (uri != null) {
+            var file = File.new_for_uri (uri);
+
+            var move_to_trash = new Gtk.MenuItem.with_label (_("Move to Trash"));
+            move_to_trash.activate.connect (() => trash ());
+
+            try {
+                var info = file.query_info (GLib.FileAttribute.ACCESS_CAN_TRASH, 0);
```

Comment:
This could be time consuming when processing large number of wallpapers, I guess you should also make this use `query_info_async`.